jib-cli/src/test/java/com/google/cloud/tools/jib/cli/buildfile/FilePropertiesStackTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.google.cloud.tools.jib.api.buildplan.FilePermissions;
+import java.time.Instant;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FilePropertiesStackTest {
+
+  @Test
+  public void testPush_simple() {
+    FilePropertiesStack testStack = new FilePropertiesStack();
+
+    testStack.push(new FilePropertiesSpec("111", "111", "1", "1", "1"));
+
+    Assert.assertEquals(FilePermissions.fromOctalString("111"), testStack.getFilePermissions());
+    Assert.assertEquals(
+        FilePermissions.fromOctalString("111"), testStack.getDirectoryPermissions());
+    Assert.assertEquals("1:1", testStack.getOwnership());
+    Assert.assertEquals(Instant.ofEpochMilli(1), testStack.getModificationTime());
+  }
+
+  @Test
+  public void testPush_stacking() {
+    FilePropertiesStack testStack = new FilePropertiesStack();
+
+    testStack.push(new FilePropertiesSpec("111", "111", "1", "1", "1"));
```

Comment:
super nit: could be useful to add inline comments for these parameters? like /*filePermissions=*/ etc.

Reply:
In general we avoid these kind of comments. Usually that kind of information is provided by IDE, you can quickly determine what each parameter is.